### DISCUSSION TITLE
Relations of the EntitySequence

### DISCUSSION THREAD

**Discussion by pponec** [vincentlauvlwj/Ktorm#425] [Q&A] [answered]
I get an error when I try to sort the `EntitySequence` by the value of the relational database table: 

> org.h2.jdbc.JdbcSQLSyntaxErrorException: Column "DEPARTMENT.NAME" not found; ...

It looks like the `EntitySequence` class API does not support entity relations.
Is it possible to ask an object type of `EntitySequence` to sort by column of a relational table according to the following pattern?

```kotlin
val result = database
    .sequenceOf(Employees(), withReferences = true)
    .sortedBy { Departments.name.asc() }
    .toList()

```

**Comment by vincentlauvlwj** [comment]
See the doc https://www.ktorm.org/en/entity-finding.html#Get-Entities-by-Sequences

![image](https://user-images.githubusercontent.com/10897214/185926039-b1790bb1-ab0f-40c7-9019-8944ed676830.png)

**Reply by pponec** [reply]
That's nice, thank you for explaining. I would like to ask about the appropriate solution for the `Employee` -> `Supervisor` cyclic binding? The following code throws a `NullPointerException` to me:

```kotlin
class Employees : Table<Employee>("employee") {
    val id = long("id").primaryKey().bindTo { it.id }
    val supervisorId = long("supervisor_id").bindTo { it.supervisor?.id }
    
    val supervisor = (supervisorId.referenceTable as Employees).aliased("supervisor") // (!)
```

**Reply by vincentlauvlwj** [reply]
See the doc https://www.ktorm.org/en/entities-and-column-binding.html#Column-Binding

**Comment by pponec** [comment, marked as answer]
I'll try to phrase the question better: how can I get a list of **employees** filtered by the **name** of their **employer** - using the `EntitySequence`? I would like to use the following query pattern:

```kotlin
val result2 = database.sequenceOf(Employees())
    .filter { (it.supervisorId.referenceTable as Employees).name eq "John" } // NPE !
    .toList()
```
The problem with this query is that the filter throws a `NullPointerException`.

**Reply by vincentlauvlwj** [reply]
Cyclic reference binding is not supported, but we have an alternative way to do this. 

First, find the employee who's name is John: 

```kotlin
val john = database.sequenceOf(Employees).find { it.name eq "John" }
```

Then, find his subordinates: 

```kotlin
val subordinates = database.sequenceOf(Employees).filter { it.supervisorId eq john!!.id }.toList()
```
